gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/domain_service/ValidateFederatedApiDomainService.java

```diff
@@ -51,7 +48,7 @@ public Api validateAndSanitizeForUpdate(final Api updateApi, final Api existingA
         );
         updateApi.setGroups(groupIds);
 
-        updateApi.setCategories(categoryDomainService.toCategoryId(existingApi, existingApi.getEnvironmentId()));
+        updateApi.setCategories(categoryDomainService.toCategoryKey(updateApi, existingApi.getEnvironmentId()));
```

Comment:
The category's `id` needs to be saved in the Database. Not the category `key`. Why change from id -> key here?
cc: @phiz71

Reply:
You are right... I make it because we expose the id instead of key to UI, but I'm must make the mapping when we read from DB and not at write time.